pkg/summary/summarizers.go

```diff
@@ -204,6 +206,22 @@ func initializeCheckErrors() {
 	logrus.Debugf("GVK Error Mapping not provided, using predefined values")
 }
 
+func checkGeneration(obj data.Object, _ []Condition, summary Summary) Summary {
+	if summary.Attributes != nil && summary.Attributes["hasObservedGeneration"] == true {
+		metadataGenerationStr := obj.String("metadata", "generation")
+		observedGenerationStr := obj.String("status", "observedGeneration")
```

Comment:
We could add `obj.ToNumber()` perhaps?

Reply:
I am using the `apimachinery` library which give feels like more appropriate like in here https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/status/generic.go#L73.

that package is better than ours.